### ISSUE TITLE
Prefix and Separator for Java-Migration Files

### ISSUE THREAD

**Issue by schneiderphilipp** [flyway/flyway#1130]
Support Prefix and Separator for Java-Migrations.

For Example:

``` java
Flyway flyway = new Flyway();
flyway.setJavaMigrationPrefix("V1_");
flyway.setJavaMigrationSeparator("_-_");
```

**Comment by axelfontaine** [comment]
Thank you for your contribution. This is however something that can already be easily achieved by creating a common migration base class that implements MigrationInfoProvider as described here: http://flywaydb.org/documentation/migration/java.html

I currently therefore do not see the need to include these additional settings in the main distribution at this time.

Cheers
Axel

**Comment by schneiderphilipp** [comment]
I thought that this would be a nice feature, because you don't need to implement something.
You are able to configurate the prefix and separator for the java migrations just like you do this for the sql migrations.

Kind regards,
Philipp

**Comment by alastairknowles** [comment]
We have a scenario where we want to use Flyway to implement two different migration strategies.

1. Simple, short running schema changes (SQL migrations)
2. Complicated, long running data changes (Java migrations)

Our strategy is to run the SQL migrations and then the Java migrations, as mixing the concerns could obviously cause problems.

We currently approach this by having a developer policy where SQL migrations have a low version number, and Java migrations have a high version number (we are working on a legacy application which has already received a number of migrations, so we cannot change the name-spacing approach for the versions to make this more robust).

Providing support for a custom version prefix for the Java migrations would provide us with a very elegant solution. We could use `S_<<version number>>` for the schema changes and `D_<<version number>>` for the data changes. We would then execute:

```
Flyway.configure()
                    ...
                    .load().migrate();
```
Twice sequentially, first specifying the prefix for the Java migrations, second the prefix for the Java migrations. Everything would be fine and dandy and we could have an error proof workflow.

I would respectfully ask you to reconsider the pull request.

We are developing this as part of a robust, idiomatic approach to versioning and maintaining very large cloud databases using Flyway (which is not a well solved problem). We are happy to blog about the solution and apportion appropriate credit, etc.

**Comment by MikielAgutu** [comment]
This issue is quite old so I'm not sure it's still relevant.

@alastairknowles Could you simply place migrations into different locations based on their 'type' (data/schema changes)?

**Comment by alastairknowles** [comment]
@MikielAgutu not quite - the desired behaviour would be to have short running and long running migrations - all of the short running migrations would always run before the long running migrations.

It's so we can fail a build if the short running migrations fail and then run the long running migrations in a detached mode.

In the end we were able to live without the behaviour by running everything in decoupled mode (which looking back at the initial comment we should blog about).

I've also just realised based upon your comments that the tool already supports this. Nothing prevents us from invoking flyway twice programatically with different migration locations - thanks for taking the time to get back to me!
